### ISSUE TITLE
How to use scheduler

### ISSUE THREAD

**Issue by tensorcopy** [Jiaming-Liu/pytorch-lr-scheduler#1]
I tried to use lr_scheduler in PyTorch but encountered an error: no zero_grad. It seems like I cannot use scheduler like optimizer. How can I use it correctly?

**Comment by Jiaming-Liu** [comment]
Assuming you are taking about the lr schedule in pytorch (master branch or 0.2.0), you should not be using it like an optimizer - - you would need to call scheduler.step() in the beginning (or ending for ReduceLROnPlateau)  of each **epoch** to update the lr, as well as optimizer.step() in each **iteration** after backwarding to update the parameters . Please follow the demo code in the document.

**Comment by tensorcopy** [comment]
Thanks for your reply! So the optimizer and scheduler are totally different and scheduler is only used for update lr. I have to use optimizer as before, like zero_grad, step, and so on.
